hikari/impl/rest.py

```diff
@@ -2425,6 +2425,42 @@ async def edit_welcome_screen(
         response = typing.cast(data_binding.JSONObject, raw_response)
         return self._entity_factory.deserialize_welcome_screen(response)
 
+    async def fetch_guild_membership_gate(
+        self, guild: snowflakes.SnowflakeishOr[guilds.PartialGuild]
+    ) -> guilds.MembershipGate:
+        route = routes.GET_GUILD_MEMBER_VERIFICATION.compile(guild=guild)
+        response = await self._request(route)
+
+        # If the member screen of a guild has never been set
+        # then Discord will respond with a 204 (no content).
```

Comment:
Hopefully we can remove this in the future. https://github.com/discord/discord-api-docs/issues/2530